File: java/lesson 2/ReverseList.java
// https://leetcode.com/problems/reverse-linked-list

class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null, next = null, current = head;        
        while (current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        return prev;
    }
}

File: kotlin/lesson 2/ReverseList.kt
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        var prev: ListNode? = null
        var next: ListNode? = null
        var current = head
        while (current != null) {
            next = current.next
            current.next = prev
            prev = current
            current = next
        }
        return prev
    }
}
